### PR TITLE
fix(CreateCategoryPopup): ensure delete button and name input stay in…

### LINKED ISSUE #3013: Delete category button is misplaced
1. create a community
2. add a category there
3. click 3 dots for category -> edit

https://user-images.githubusercontent.com/82375995/126555049-bde9d8f5-7ba4-4466-96e2-05101e734ac5.mov

Screenshot 2021-07-21 at 23 24 44

### PR DESCRIPTION
**Pull request by PascalPrecht**
… viewport

There are a few issues with the popup's content when there are many channels in a community:

1. Channel items belonging to other categories would still take space in the list
1. Due to the longer list of channels, the delete button wouldn't be fully visible (scrolling down
   helps here)
2. Scrolling down is hard because the `ListView` in use is `interactive`
3. Even when `interactive` is set to false, one would scroll the name input out of the viewport.

To solve these, this commit rearchitects the popup's content such that:

1. The name input is always static and in place
2. The scrollview starts with the channel list and ends with the channel list
3. The delete button is positioned below the scrollview
4. The scrollview has a max height of 300 so that the popup doesn't grow too big
5. Invisible channel items won't have a height anymore

The result is that the middle section of the popup becomes scrollable in case there's many
channel items, while both, the name input and the delete button stay in the viewport.

Fixes #3013

### CODE REVIEW

File: ui/app/AppLayouts/Chat/CommunityComponents/CreateCategoryPopup.qml [hunk 1]
```diff
@@ -48,166 +48,173 @@ StatusModal {
...
-        clip: true
+        Item {
+            width: parent.width
+            height: 76
+            Input {
+                id: nameInput
+                width: parent.width -32
+
+                anchors.centerIn: parent
+                anchors.left: undefined
```
Comment (pinebit) [inline ui/app/AppLayouts/Chat/CommunityComponents/CreateCategoryPopup.qml:64]: interesting... why setting these to `undefined`?

**Review by pinebit — APPROVED** [review]

File: ui/app/AppLayouts/Chat/CommunityComponents/CreateCategoryPopup.qml (hunk 1 above)
Comment (PascalPrecht) [reply ui/app/AppLayouts/Chat/CommunityComponents/CreateCategoryPopup.qml:64]: This unfortunately is a rather big flaw in the API of `Input` (hopefully fixed soon once we move to `StatusInput` provided by StatusQ).

`Input` comes with `anchors.left: parent.left` and `anchors.right: parent.right` by default. In this context however, I need to set a custom width (L61) but QML doesn't allow me to because the component is already anchored left in right (taking the entire width).

So basically, I need to **undo** default behaviour to make it work for me. Again, this will be fixed when we switch to the (not yet existing) `StatusInput` which won't rely on `parent` by itself.

Hope this makes sense :D

Also, should add that that particular change is unrelated to this PR

**Review by alaibe — APPROVED** [review]

**Comment by anastasiyaig** [comment]
Thank you for fixing it @PascalPrecht

**Review by jrainville — APPROVED** [review]
